main/src/com/google/refine/importers/tree/XmlImportUtilities.java

```diff
@@ -356,23 +357,29 @@ static protected void findRecord(
         String fullName = composeName(parser.getPrefix(), localName);
         if (recordPathSegment.equals(localName) || recordPathSegment.equals(fullName)) {
             if (pathIndex < recordPath.length - 1) {
-                while (parser.hasNext() && limit != 0) {
-                    Token eventType = parser.next();
-                    if (eventType == Token.StartEntity) {
-                        findRecord(project, parser, recordPath, pathIndex + 1, rootColumnGroup, limit--,
-                                trimStrings, storeEmptyStrings, guessDataTypes);
-                    } else if (eventType == Token.EndEntity) {
-                        break;
-                    } else if (eventType == Token.Value) {
-                        // This is when the user picks a specific field to import, not a whole object or element.
-                        if (pathIndex == recordPath.length - 2) {
-                            String desiredFieldName = recordPath[pathIndex + 1];
-                            String currentFieldName = parser.getFieldName();
-                            if (desiredFieldName.equals(currentFieldName)) {
-                                processFieldAsRecord(project, parser, rootColumnGroup, trimStrings, storeEmptyStrings, guessDataTypes);
+                try {
+                    while (parser.hasNext() && limit != 0) {
+                        Token eventType = parser.next();
+                        if (eventType == Token.StartEntity) {
+                            findRecord(project, parser, recordPath, pathIndex + 1, rootColumnGroup, limit--,
+                                    trimStrings, storeEmptyStrings, guessDataTypes);
+                        } else if (eventType == Token.EndEntity) {
+                            break;
+                        } else if (eventType == Token.Value) {
+                            // This is when the user picks a specific field to import, not a whole object or element.
+                            if (pathIndex == recordPath.length - 2) {
+                                String desiredFieldName = recordPath[pathIndex + 1];
+                                String currentFieldName = parser.getFieldName();
+                                if (desiredFieldName.equals(currentFieldName)) {
+                                    processFieldAsRecord(project, parser, rootColumnGroup, trimStrings, storeEmptyStrings, guessDataTypes);
+                                }
                             }
                         }
                     }
+                } catch (TreeReaderException e) {
+                    logger.error("Exception from parse", e);
+                    throw new TreeReaderException("Exception from parsing: " + e.getLocalizedMessage());
```

Comment:
It is not completely clear to me what value there is in catching and rethrowing the exception. Why do we need to log it since it is shown to the user anyway?

Reply:
@wetneb Sure there is no need to log the error. The value of rethrowing the exception is to present the error message on the UI since it will not be logged to the console

Reply:
then how about not catching the exception at all?

Reply:
Not catching the exception is what caused this issue that the PR is trying to fix

Reply:
I think there is a work around in not catching the error and hadling it from the UI via the recived response payload. Let me try it out

Reply:
Not catching the exception and presenting the error messge from the front end works fine